### PR TITLE
Remote handlers maintained for remote-parser 3

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/remote-handlers/abstract.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import eventToPromise from 'event-to-promise'
 import through2 from 'through2'
+import assign from 'lodash.assign'
```
Comment (julien-f) [inline src/remote-handlers/abstract.js]: Sorted imports.

File: src/remote-handlers/abstract.js [hunk 2]
```diff
@@ -15,7 +16,7 @@ import {
 
 export default class RemoteHandlerAbstract {
   constructor (remote) {
-    this._remote = parse({...remote})
+    this._remote = assign({...remote}, parse(remote.url))
```
Comment (julien-f) [inline src/remote-handlers/abstract.js]: You can use `{ ...remote, ...parse(remote.url) }`, assign is not needed here.

File: src/remote-handlers/local.js [hunk 3]
```diff
@@ -12,16 +12,20 @@ import {
 
 export default class LocalHandler extends RemoteHandlerAbstract {
   get type () {
-    return 'local'
+    return 'file'
```
Comment (julien-f) [inline src/remote-handlers/local.js]: Does that make sense to you? I do not want to push a change you're uncomfortable with.

File: src/remote-handlers/local.js [hunk 4]
```diff
@@ -30,8 +34,8 @@ export default class LocalHandler extends RemoteHandlerAbstract {
   async _sync () {
     if (this._remote.enabled) {
       try {
-        await fs.ensureDir(this._remote.path)
-        await fs.access(this._remote.path, fs.R_OK | fs.W_OK)
+        await fs.ensureDir(this._getRealPath())
```
Comment (julien-f) [inline src/remote-handlers/local.js]: Shouldn't it be `_getFilePath()`?

File: src/remote-handlers/local.js [hunk 5]
```diff
@@ -12,16 +12,20 @@ import {
...
+    return 'file'
+  }
+
+  _getRealPath () {
+    return this._remote.path
   }
 
   _getFilePath (file) {
-    const parts = [this._remote.path]
+    const parts = [this._getRealPath()]
```
Comment (julien-f) [inline src/remote-handlers/local.js]: Use a `const` if you're gonna use it multiple time?

File: src/remote-handlers/local.js (hunk 4 above)
Comment (julien-f) [inline src/remote-handlers/local.js]: Oh, I think I understand now.

File: src/remote-handlers/local.js (hunk 3 above)
Comment (marsaud) [inline src/remote-handlers/local.js]: No problem for me, "local" is a user UI feeling name, in the code we do talk about xoa files.

File: src/remote-handlers/local.js (hunk 4 above)
Comment (marsaud) [inline src/remote-handlers/local.js]: The 2 are clear to me, I meant, "the path on this very local concrete file system" :p

File: src/remote-handlers/local.js (hunk 5 above)
Comment (marsaud) [inline src/remote-handlers/local.js]: What do you mean ?...

Comment (julien-f) [inline src/remote-handlers/local.js]: You call twice this method in the same scope, you can use a constant value instead.

File: src/remote-handlers/local.js (hunk 4 above)
Comment (marsaud) [inline src/remote-handlers/local.js]: Ah ok, I did not catch the meaning of your question !

Comment (marsaud) [inline src/remote-handlers/local.js]: Yes that's it, getRealPath is a polymorphic (nfs extends local) method to locate the remote point (mounted or not) on the file system, getFilePath gives the "true" path to a file supposed to be in the remote point.

**Comment by julien-f** [comment]
I'll merge this afet xo-remote-parser has been published.

File: src/remote-handlers/nfs.js [hunk 6]
```diff
@@ -47,7 +47,7 @@ export default class NfsHandler extends LocalHandler {
 
   async _mount () {
     await fs.ensureDir(this._getRealPath())
-    return execa('mount', ['-t', 'nfs', '-o', 'vers=3', `${this._remote.host}:/${this._remote.path}`, this._getRealPath()])
+    return execa('mount', ['-t', 'nfs', '-o', 'vers=3', `${this._remote.host}:${this._remote.path}`, this._getRealPath()])
```
Comment (julien-f) [inline src/remote-handlers/nfs.js]: It looks better like that :)

**Comment by Danp2** [comment]
@julien-f Shouldn't the NFS handler be using `this._getRealPath()` instead of `remote.path` during the attempt to unmount an NSF share?

**Comment by julien-f** [comment]
You're right, thanks!

**Pull request merged**